mtp_cashbook/apps/disbursements/views.py

```diff
@@ -555,13 +563,22 @@ def dispatch(self, request, **kwargs):
             self.disbursement = self.api_session.get(
                 'disbursements/{pk}/'.format(pk=kwargs['pk'])
             ).json()
-            self.disbursement['remittance'] = 'yes' if self.disbursement.get('remittance_description') else 'no'
         except HttpNotFoundError:
             raise Http404('Disbursement %s not found' % kwargs['pk'])
         for view in list(self.get_previous_views()) + [self.__class__]:
             if not issubclass(view, BasePagedFormView) or not view.is_form_enabled(self.valid_form_data):
                 continue
-            view.form_class.serialise_data(request.session, self.disbursement)
+            disbursement_data = self.disbursement.copy()
+            if disbursement_data.get('recipient_is_company'):
+                disbursement_data['recipient_type'] = 'company'
+                disbursement_data['recipient_company_name'] = disbursement_data['recipient_last_name']
+                disbursement_data['recipient_first_name'] = ''
+                disbursement_data['recipient_last_name'] = ''
+            else:
+                disbursement_data['recipient_type'] = 'person'
+                disbursement_data['recipient_company_name'] = ''
+            disbursement_data['remittance'] = 'yes' if disbursement_data.get('remittance_description') else 'no'
```

Comment:
This should probably also be in the form class or view subclass. It's in a loop here, so it's going to be doing it over and over again which seems weird.